icu4c/source/i18n/measunit_extra.cpp

```diff
@@ -422,24 +460,28 @@ class Parser {
                     break;
 
                 case COMPOUND_PART_TIMES:
-                    if (fAfterPer) {
-                        result.dimensionality = -1;
+                    if (previ == 0) {
+                        // Can't start with "-".
+                        status = kUnitIdentifierSyntaxError;
+                        return;
                     }
+                    if (fAfterPer) { result.dimensionality = -1; }
```

Comment:
I would put this on separate lines.  It's easier when reading code to see this when it's on its own line.